vc_zoom/indico_vc_zoom/forms.py

```diff
@@ -0,0 +1,110 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2020 CERN and ENEA
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+from flask import session
+
+from wtforms.fields.core import BooleanField, StringField
+from wtforms.fields.simple import HiddenField, TextAreaField
+from wtforms.validators import DataRequired, ValidationError
+
+from indico.modules.vc.forms import VCRoomAttachFormBase, VCRoomFormBase
+from indico.util.user import principal_from_identifier
+from indico.web.forms.base import generated_data
+from indico.web.forms.fields import IndicoRadioField, PrincipalField
+from indico.web.forms.validators import HiddenUnless, IndicoRegexp
+from indico.web.forms.widgets import SwitchWidget
+
+from indico_vc_zoom import _
+
+
+class VCRoomAttachForm(VCRoomAttachFormBase):
+    password_visibility = IndicoRadioField(_("Passcode visibility"),
+                                           description=_("Who should be able to know this meeting's passcode"),
+                                           orientation='horizontal',
+                                           choices=[
+                                               ('everyone', _('Everyone')),
+                                               ('logged_in', _('Logged-in users')),
+                                               ('no_one', _("No one"))])
+
+
+class VCRoomForm(VCRoomFormBase):
+    """Contains all information concerning a Zoom booking."""
+
+    advanced_fields = {'mute_audio', 'mute_host_video', 'mute_participant_video'} | VCRoomFormBase.advanced_fields
+
+    skip_fields = advanced_fields | VCRoomFormBase.conditional_fields
+
+    meeting_type = IndicoRadioField(_("Meeting Type"),
+                                    description=_("The type of Zoom meeting to be created"),
+                                    orientation='horizontal',
+                                    choices=[
+                                        ('regular', _('Regular Meeting')),
+                                        ('webinar', _('Webinar'))])
+
+    host_choice = IndicoRadioField(_("Meeting Host"), [DataRequired()],
+                                   choices=[('myself', _("Myself")), ('someone_else', _("Someone else"))])
+
+    host_user = PrincipalField(_("User"),
+                               [HiddenUnless('host_choice', 'someone_else'), DataRequired()])
+
+    password = StringField(_("Passcode"),
```

Comment:
Should we use `passcode` instead of `password` for the internal names? Or is it `password` in the Zoom APIs as well?

Reply:
It's password in the API as well. That's why I left it.